lib/logstash/dead_letter_post_office.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+class LogStash::DeadLetterPostOffice
+
+  def self.logger=(logger)
+    @logger = logger
+  end
+
+  def self.logger
+    @logger ||= Cabin::Channel.get(LogStash)
+  end
+
+  def self.destination=(destination)
+    logger.info("Dead letter events will be sent to \"#{destination.location}\".")
+    @destination = destination
+  end
+
+  def self.<<(events)
+    events = [events] unless events.is_a?(Array)
+
+    events.each do |event|
+      logger.warn("dead letter received!", :event => event.to_hash)
+      event.tag("_dead_letter")
+      event.cancel
+      @destination << event
+    end
+  end
+
+  module Destination
+
+    class Base
+      def location; end
+      def <<(event); end
+    end
+
+    class File < Base
+
+      START_TIME = Time.now
```

Comment:
This dead letter mechanism v0.1 is quite dumb: save "lost" events to a file. I set up this destination to include an approximate start up time of Logstash so there's 1 dead letter file per start up.

Having a constant file name here would make it hard to understand which events were lost during each execution, though I might be overthinking this.